### ISSUE TITLE
21ICLR | An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale

### ISSUE THREAD

**Issue by XFeiF** [XFeiF/Video_PaperNotes#23]
[Paper](https://arxiv.org/abs/2010.11929) & [Code-pytorch](https://github.com/lucidrains/vit-pytorch)  

![Image](https://github.com/lucidrains/vit-pytorch/raw/main/vit.png)

**Comment by XFeiF** [comment]
## TEN QUESTIONS
**1. What is the problem addressed in the paper?**
Attention can be used without relying on CNNs. A pure transformer applied directly to sequences of image patches can perform very well on image classification tasks. When pre-trained on large amounts of data and transferred to multiple mid-sized or small image recognition benchmarks(ImageNet, CIFAR-100, VTAB, etc), Vision-Transformer(ViT) attains excellent results compared to state-of-the-art convolutional networks while requiring substantially fewer computational resources to train.  

Vit Algorithm:
The algorithm is shown in the figure above. 
1. Split an image info fixed-sized patches.  
2. Linearly embed each of them. 
3. Add position embeddings.   
4. To perform classification, they use the standard approach of adding an extra learnable "classification token" to the sequence.    
5. Feed the resulting sequence of vectors to a standard Transformer encoder (times by L).   
6. MLP Head for classification.  
The transformer encoder is constructed by ( layer norm->attention->dropout->identity->layer norm->mlp->identity ).   
The arch of MLP block is (dense->gelu(gaussian error linear units)->dropout->dense->dropout). 

The classification head is implemented by a MLP with one hidden layer at pre-training time and by a single linear layer at fine-tuning time.

**2. Is this a new problem?**
ViT is inspired by the Transformer scaling success in NLP. So, it is not an entirely new problem. But it demonstrates that Vision Transformer (attention without relying on CNNs) matches or exceeds the state of the art on many image classification datasets.

**3. What is the scientific hypothesis that the paper is trying to verify?**
Attention can be used without relying on CNNs.  A pure transformer applied directly to sequences of image patches can perform very well on image classification tasks. 

**4. What are the key related works and who are the key people working on this topic?**  
Key related works: Transformer.  
Key people: the authors(Alexey Dosovitskiy, Lucas Beyer, etc), Vaswani etc.

**5. What is the key to the proposed solution in the paper?**
The key is how to get the embedded patches before sending them to the transformer encoder. The authors proposed to split an image into fixed-size patches, linearly embed each of them, add position embeddings to the resulting sequence of vectors. And follow the former approach of adding an extra learnable "classification token" to the sequence.  

**6. How are the experiments designed?**   
1. ResNet v.s. ViT v.s. BiT v.s. Noisy Student.
2. Comparison with the SoTA on image classification benchmarks (acc, mean with std). ViT trained on JFT.
3. Ablation study on different size datasets. (ImageNet-10K, ImageNet-21K, CIFAR-100, VTAB, JFT-300M).
4. Scaling study.( ResNet v.s. ViT v.s. Hybrid)
5. Inspecting vision transformer.

**7. What datasets are built/used for the quantitative evaluation? Is the code open-sourced?**   
Training: ImageNet-10K, ImageNet-21K, CIFAR-100, VTAB, JFT-300M
Metrics: fine-tuning accuracy.
Code available.

**8. Is the scientific hypothesis well supported by evidence in the experiments?**  
Attention can be used without relying on CNNs. A pure transformer(ViT) applied directly to sequences of image patches can perform very well on image classification tasks. 

**9. What are the contributions of the paper?**
1. Attention can be used without relying on CNNs. A pure transformer applied directly to sequences of image patches can perform very well on image classification tasks. 
2. Only when pre-trained on large amounts of data and transferred to multiple mid-sized or small image recognition benchmarks(ImageNet, CIFAR-100, VTAB, etc), Vision-Transformer(ViT) attains excellent results compared to state-of-the-art convolutional networks while requiring substantially fewer computational resources to train.   

**10. What should/could be done next?**  
Give your answers?

**Comment by XFeiF** [comment]
暴力美学？

**Comment by XFeiF** [comment]
Can the model be applied to video classification?  
- We do not have enough data.
- Image patch -> Frame sequences?
- The position embedding is very interesting if we can embed frame index into video classification...
